### PR TITLE
State-tracker

### PR DESCRIPTION
**Pull request by Frankonly**
This is the state-tracker version of iotex-core who can track state changes like balance changes. The Tracker is written in stateDB, who records particular state changes when server is handling a new block and stores them into datebase as working set is committed into stateDB.

To run the state-tracker, we need to specify MySQL connection string and datababse name:
```
export CONNECTION_STRING=username:password@protocol(address)/
export DB_NAME=dbname
```

### CODE REVIEW

File: state/tracker/tracker.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	}
+}
+
+// Append appends new state change
+func (t *stateTracker) Append(c StateChange) {
+	t.changes = append(t.changes, c)
+}
+
+// Snapshot records current status of changes
+func (t *stateTracker) Snapshot() {
```
Comment (zjshen14) [inline state/tracker/tracker.go]: What the meaning of the snapshot?

File: db/sql/analyticssql/mysql_test.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+package sql
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+)
+
+const (
+	connectStr = "ba8df54bd3754e:9cd1f263@tcp(us-cdbr-iron-east-02.cleardb.net:3306)/"
```
Comment (zjshen14) [inline db/sql/analyticssql/mysql_test.go]: Don’t leave this in the codebase

File: db/sql/analyticssql/storebase_tests.go [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+	"database/sql"
+	"errors"
+	"testing"
+
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/stretchr/testify/require"
+)
+
+// ActionHistory define the schema for action history
+type ActionHistory struct {
```
Comment (zjshen14) [inline db/sql/analyticssql/storebase_tests.go]: Is this copied from previous code of action index?

File: state/tracker/tracker.go (hunk 1 above)
Comment (Frankonly) [reply state/tracker/tracker.go]: snapshot the cached state changes when the workingset calls Snapshot.

File: db/sql/analyticssql/storebase_tests.go (hunk 3 above)
Comment (Frankonly) [reply db/sql/analyticssql/storebase_tests.go]: Yes

Comment (Frankonly) [reply db/sql/analyticssql/storebase_tests.go]: To avoid cycle import

**Pull request merged**